documentation/book/security.adoc

```diff
@@ -0,0 +1,64 @@
+== Security
+
+The Apache Kafka project supports data encryption by means of the SSL/TLS protocol.
+This makes it possible to encrypt data transferred between brokers (interbroker communication) and between clients and brokers.
+Leveraging the SSL/TLS support, it is also possible to have mutual authentication, where the Kafka broker authenticates the client by verifying its certificate, and the client does likewise with the server's certificate.
+
+The Cluster Operator sets up the SSL/TLS certificates to provide this encryption and authentication.
+
+=== Certificates
+
+Each Kafka broker needs its own private and public keys in order to support encryption.
+The public key has to be signed by a certificate authority (CA) in order to have a related X.509 certificate for providing server authentication and encrypting the communication channel with the client (which could be another broker as well).
+More precisely, each broker has two different certificates (and related private keys), one for interbroker communication  and one for communication with clients.
+These broker certificates are signed using different CAs: The "internal-ca" signs the certificates used for interbroker communication, and the "clients-ca" signs the certificates used for client communication.
+The CAs themselves use self-signed certificates.
```

Comment:
CA is always self-signed. So I'm, not sure we need to mention this here.

Reply:
Well in a PKI you can have different CA certificates at different levels. The first one is self-signed but the other ones down the chain could be signed by the previous one (so not be self-signed), or ?

Reply:
Yeah, but if you say CA, I expect the root certificate ... and that is always self-signed if its a root, right? Anyway ... I don't mind this, it just seemed strange.